### PR TITLE
support test tiflash in case bank/bank2/crud/ledger/sqllogic

### PR DESCRIPTION
**Pull request by lidezhu**
### What problem does this PR solve? 
support test tiflash in case bank/bank2/crud/ledger/sqllogic

### Check List 

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Review by mahjonp — COMMENTED** [review]
I only look up the changes on`pkg/test-infra`, LGTM

File: testcase/sqllogictest/pkg/util/util.go [hunk 1]
```diff
@@ -225,3 +225,40 @@ func RandomlyChangeReplicaRead(job, replicaRead string, db *sql.DB) {
...
+		if available == 0 {
+			return errors.Errorf("TiFlash replica not available")
+		}
+		if available == -1 {
+			return errors.Errorf("query TiFlash replica status failed.")
+		}
+		return nil
+	})
+	if err != nil {
+		return errors.Errorf("wait TiFlash replica of %v.%v error after %v seconds: %d", dbName, tableName, retryCount, err)
```
Comment (lilinghai) [inline testcase/sqllogictest/pkg/util/util.go:261]: the format of err may be "%s" or "%v"

File: util/util.go [hunk 2]
```diff
@@ -225,3 +225,40 @@ func RandomlyChangeReplicaRead(job, replicaRead string, db *sql.DB) {
...
+		if available == 0 {
+			return errors.Errorf("TiFlash replica not available")
+		}
+		if available == -1 {
+			return errors.Errorf("query TiFlash replica status failed.")
+		}
+		return nil
+	})
+	if err != nil {
+		return errors.Errorf("wait TiFlash replica of %v.%v error after %v seconds: %d", dbName, tableName, retryCount, err)
```
Comment (lilinghai) [inline util/util.go:261]: ditto

File: testcase/sqllogictest/pkg/util/util.go [hunk 3]
```diff
@@ -225,3 +225,40 @@ func RandomlyChangeReplicaRead(job, replicaRead string, db *sql.DB) {
...
+	}
+	log.Infof("begin to execute %s", sql)
+	MustExec(db, sql)
+
+	err := RunWithRetry(ctx, retryCount, time.Second, func() error {
+		instance, available := "", -1
+		if err := db.QueryRow("select INSTANCE FROM information_schema.cluster_info where type='tidb'").Scan(&instance); err != nil {
+			return err
+		}
+		query := fmt.Sprintf("select AVAILABLE from information_schema.tiflash_replica where "+
```
Comment (lilinghai) [inline testcase/sqllogictest/pkg/util/util.go:245]: if the dbName is "", the TABLE_SCHEMA condition should be removed.

File: util/util.go [hunk 4]
```diff
@@ -225,3 +225,40 @@ func RandomlyChangeReplicaRead(job, replicaRead string, db *sql.DB) {
...
+	}
+	log.Infof("begin to execute %s", sql)
+	MustExec(db, sql)
+
+	err := RunWithRetry(ctx, retryCount, time.Second, func() error {
+		instance, available := "", -1
+		if err := db.QueryRow("select INSTANCE FROM information_schema.cluster_info where type='tidb'").Scan(&instance); err != nil {
+			return err
+		}
+		query := fmt.Sprintf("select AVAILABLE from information_schema.tiflash_replica where "+
```
Comment (lilinghai) [inline util/util.go:245]: ditto

File: testcase/sqllogictest/sqllogic.go [hunk 5]
```diff
@@ -431,6 +454,16 @@ func (t *tester) execStatement(ctx context.Context, stmt statement) error {
 		}
 		return fmt.Errorf("%s: expected success, but found %v", stmt.pos, err)
 	}
+
+	if tiFlashDataReplicas > 0 {
+		// grep create table statement and set TiFlash replica
+		// replace '\n' with ' ' so that we can get table name for multi-line create statement
+		if m := createTableRE.FindStringSubmatch(strings.Replace(stmt.sql, "\n", " ", -1)); m != nil {
+			if err := util.SetAndWaitTiFlashReplica(ctx, t.db, "", m[1], tiFlashDataReplicas, 3*dbTryNumber); err != nil {
```
Comment (lilinghai) [inline testcase/sqllogictest/sqllogic.go:462]: maybe we should indicate the dbName.

**Review by lilinghai — COMMENTED** [review]
all tests can run normally except sqllogic. Alter fixed , I will execute again with the http://fileserver.pingcap.net/download/builds/pingcap/case-data/sqllogic/sqllogictest_case.tar.gz

File: testcase/sqllogictest/pkg/util/util.go (hunk 1 above)
Comment (lidezhu) [reply testcase/sqllogictest/pkg/util/util.go]: fixed.

File: util/util.go (hunk 2 above)
Comment (lidezhu) [reply util/util.go]: fixed.

File: testcase/sqllogictest/pkg/util/util.go (hunk 3 above)
Comment (lidezhu) [reply testcase/sqllogictest/pkg/util/util.go:241]: fixed.

File: util/util.go (hunk 4 above)
Comment (lidezhu) [reply util/util.go:241]: fixed.

File: testcase/sqllogictest/sqllogic.go (hunk 5 above)
Comment (lidezhu) [reply testcase/sqllogictest/sqllogic.go]: fixed.

**Review by lilinghai — APPROVED** [review]
lgtm

**Pull request merged**